File: android/src/main/kotlin/com/idavize/flutter_iadvize_sdk/OnReceiveMessageStreamHandler.kt
package com.idavize.flutter_iadvize_sdk

import android.os.Handler
import io.flutter.plugin.common.EventChannel
import java.util.*

class OnReceiveMessageStreamHandler: EventChannel.StreamHandler {
    var sink: EventChannel.EventSink? = null

    fun onMessage(message: String) {
        sink?.success(message)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}

File: ios/Classes/OnReceiveMessageStreamHandler.swift
import Foundation
import Flutter

class OnReceiveMessageStreamHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    @objc func onMessage(message: String) {
        guard let sink = sink else { return }
        sink(message)
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}